pkg/storegateway/bucket.go

```diff
@@ -1787,8 +1718,10 @@ func storeCachedLabelValues(ctx context.Context, indexCache indexcache.IndexCach
 
 // bucketBlockSet holds all blocks.
 type bucketBlockSet struct {
-	mtx    sync.RWMutex
-	blocks []*bucketBlock // Blocks sorted by mint, then maxt.
+	// mtx protects the below data strcutures, helping to keep them in sync.
+	mtx      sync.RWMutex
+	blockSet sync.Map       // Maps block's identifier to the *bucketBlock projection.
```

Comment:
```suggestion
	blockSet sync.Map       // Maps block's ulid.ULID to the *bucketBlock.
```

isn't a map and a projection the same thing?